core/trino-main/src/test/java/io/trino/execution/TestAddColumnTask.java

```diff
@@ -124,6 +151,146 @@ public void testAddColumnWithComment()
                                 .build());
     }
 
+    @Test
+    void testAddDefaultColumn()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), FAIL);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).orElseThrow();
+        assertThat(metadata.getTableMetadata(testSession, table).columns())
+                .containsExactly(new ColumnMetadata("test", BIGINT));
+
+        ColumnDefinition column = new ColumnDefinition(
+                new NodeLocation(1, 1),
+                QualifiedName.of("new_col"),
+                toSqlType(INTEGER),
+                Optional.of(new LongLiteral(new NodeLocation(1, 1), "123")),
+                true,
+                ImmutableList.of(),
+                Optional.empty());
+
+        getFutureValue(executeAddColumn(asQualifiedName(tableName), column, new ColumnPosition.Last(), false, false));
+
+        List<ColumnMetadata> columns = metadata.getTableMetadata(testSession, table).columns();
+        assertThat(columns).hasSize(2);
+
+        assertThat(columns.get(0).getName()).isEqualTo("test");
+
+        assertThat(columns.get(1).getName()).isEqualTo("new_col");
+        assertThat(columns.get(1).getDefaultValue()).contains(NullableValue.of(INTEGER, 123L));
+    }
+
+    @Test
+    void testAddDefaultColumnTypeCoercion()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), FAIL);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).orElseThrow();
+        assertThat(metadata.getTableMetadata(testSession, table).columns())
+                .containsExactly(new ColumnMetadata("test", BIGINT));
+
+        ColumnDefinition column = new ColumnDefinition(
+                new NodeLocation(1, 1),
+                QualifiedName.of("new_col"),
+                toSqlType(createVarcharType(4)),
+                Optional.of(new StringLiteral(new NodeLocation(1, 1), "abcde")),
```

Comment:
Shouldn't this fail? 
See the javadoc for `LogicalPlanner.noTruncationCast()`.
Or will it fail when we effectively want to insert the default value?

Reply:
https://github.com/trinodb/trino/pull/25679#discussion_r2086733106